### PR TITLE
[MODDATAIMP-947] Fix improper system user account management

### PR DESCRIPTION
**Pull request by ncovercash**
# [Jira MODDATAIMP-947](https://issues.folio.org/browse/MODDATAIMP-947)

## Purpose
With Poppy, certain changes were made to user logins/token management.  We updated the code here to handle this, however, missed a couple spots relating to password setting on tenant install.

## Approach
Ensure proper handling of login results when installing on tenant

### CODE REVIEW

**Review by KaterynaSenchenko — APPROVED** [review]

File: src/main/java/org/folio/rest/impl/ModTenantAPI.java [hunk 1]
```diff
@@ -46,10 +46,14 @@ public ModTenantAPI() {
...
   @Override
   Future<Integer> loadData(TenantAttributes attributes, String tenantId, Map<String, String> headers, Context context) {
-    if (fileSplittingEnabled) {
-      systemUserAuthService.initializeSystemUser(headers);
-    }
-    return super.loadData(attributes, tenantId, headers, context)
+    return Future.succeededFuture()
+      .compose(v -> {
+        if (fileSplittingEnabled) {
+          return systemUserAuthService.initializeSystemUser(headers);
```
Comment (okolawole-ebsco) [inline src/main/java/org/folio/rest/impl/ModTenantAPI.java:52]: I think the system user should be created everytime. If a tenant is installed with file splitting disabled, it cannot have DI working when file splitting is enabled.

**Review by RomanChernetskyi — APPROVED** [review]

File: src/main/java/org/folio/rest/impl/ModTenantAPI.java (hunk 1 above)
Comment (ncovercash) [reply src/main/java/org/folio/rest/impl/ModTenantAPI.java:52]: Our goal was to not have file splitting be a breaking change/only opt-in; IMO having to set its password environment variable is a breaking change.

I'll add a note to the release notes though; in generally, I'd advise calling install whenever changing configuration like this.  The system user's password env variable would have the same issue, where an install is required each time

Comment (ncovercash) [reply src/main/java/org/folio/rest/impl/ModTenantAPI.java:52]: Also, I don't want to spook admins with a new user appearing out of the blue

Comment (okolawole-ebsco) [reply src/main/java/org/folio/rest/impl/ModTenantAPI.java:52]: Sounds good!

**Pull request merged**